src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +35,12 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        String[] addressDetails = trimmedAddress.split(", ");
+        block = new Block(addressDetails[INDEX_OF_BLOCK]);
+        street = new Street (addressDetails[INDEX_OF_STREET]);
+        unit = new Unit (addressDetails[INDEX_OF_UNIT]);	
+        postalCode = new PostalCode (addressDetails[INDEX_OF_POSTALCODE]);
+        this.value = block.getBlock()+", "+street.getStreet()+", "+unit.getUnit()+", "+postalCode.getPostalCode();
```

Comment:
consider doing this in the toString() method. If not there can be duplicate inconsistent data right?